### PR TITLE
feat: Make popup height configurable

### PR DESCRIPTION
**Pull request by wilriker**
As of now only popup width is configurable (via the undocumented config option `popup_width`). This PR adds the ability to also control popup height (via also undocumented config option `popup_height`). In addition it honors available window estate.

### CODE REVIEW

**Comment by crispgm** [comment]
Thanks. The idea looks good to me. And the `actual_content_height` is a nice tweak on UI.

Please run the `make format` to keep the code style. I will review later.

File: lua/go/output.lua [hunk 1]
```diff
@@ -39,27 +39,35 @@ function M.show_job_error(prefix, code, results)
...
-        if cc >= width then
-            width = cc
+        if cc >= pos_opts.width then
+            pos_opts.width = cc
         end
     end
-    -- min width= 80
-    if width < 80 then
-        width = 80
+	if config.options.popup_height then
```
Comment (crispgm) [inline lua/go/output.lua]: Please put the default width/height in `lua/go/config.lua`.

File: lua/go/output.lua [hunk 2]
```diff
@@ -86,16 +94,19 @@ function M.popup_job_result(results, opts)
     local buf_nr = vim.api.nvim_create_buf(false, true)
     vim.api.nvim_buf_set_option(buf_nr, 'bufhidden', 'wipe')
     vim.api.nvim_buf_set_lines(buf_nr, 0, -1, true, results)
+    local actual_content_height = vim.api.nvim_buf_line_count(buf_nr)
     local title = opts.title
-    local top, width = opts.top or 1, opts.width or 80
+    local pos_opts = opts.pos_opts
+	local top = pos_opts.win_height - (math.min(pos_opts.height, actual_content_height) + 1)
```
Comment (crispgm) [inline lua/go/output.lua]: I think we may retrieve `win_height` again with `vim.fn.winheight(0)`. Pass with function return value is weird here.

File: lua/go/output.lua (hunk 1 above)
Comment (crispgm) [reply lua/go/output.lua]: And also in README. But I can do that for you.

Comment (wilriker) [reply lua/go/output.lua]: Put into defaults and README.

File: lua/go/output.lua (hunk 2 above)
Comment (wilriker) [reply lua/go/output.lua]: Removed from returned object and fetched again before creating popup.

**Comment by wilriker** [comment]
Thanks!

**Pull request merged**